assets/terraform-modules/bare-metal/flatcar-linux/kubernetes/variables.tf

```diff
@@ -240,6 +240,12 @@ variable "install_pre_reboot_cmds" {
   description = "shell commands to execute on the provisioned host after installation finished and before reboot, e.g., docker run --privileged --net host --rm debian sh -c 'apt update && apt install -y ipmitool && ipmitool chassis bootdev disk options=persistent'"
 }
 
+variable "ignore_worker_changes" {
```

Comment:
If this is causing troubles, perhaps we can turn the logic, so re-provisioning is optional, as on other platforms, but it's opt-in for interested parties?

This will also prevent accidental breakage on existing clusters while upgrading, if user forgets to set a specific config option.

We could also think about extending that feature, to only allow certain domains to be re-provisioned.

Reply:
We can change the default, yes. Is this consistent for all platforms?

Reply:
(It's not really causing troubles unless the user has important data in storage volumes)

Reply:
I think it's consistent. At least on Packet we do not replace nodes during upgrades.

Reply:
@invidian  @pothos 

I've revered the behavior. It ignores the changes by default.

Reply:
> We could also think about extending that feature, to only allow certain domains to be re-provisioned.

We can think about it as a new feature. Probably out of scope for this PR.

Reply:
@invidian Default is `true` , i.e it ignores the worker changes and doesn't reprovision.

If one wants re-provisioning of the workers the user must set `ignore_worker_changes = false`

Reply:
But this is default in Terraform, which will be overwritten by Go template, as zero value of bool is `false` in Go.

Reply:
@invidian updated.